pkg/kubelet/stats/pidlimit/pidlimit_linux.go

```diff
@@ -25,19 +25,25 @@ import (
 	"syscall"
 	"time"
 
-	"k8s.io/apimachinery/pkg/apis/meta/v1"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	statsapi "k8s.io/kubelet/pkg/apis/stats/v1alpha1"
 )
 
 // Stats provides basic information about max and current process count
 func Stats() (*statsapi.RlimitStats, error) {
 	rlimit := &statsapi.RlimitStats{}
 
-	if content, err := ioutil.ReadFile("/proc/sys/kernel/pid_max"); err == nil {
-		if maxPid, err := strconv.ParseInt(string(content[:len(content)-1]), 10, 64); err == nil {
-			rlimit.MaxPID = &maxPid
+	var taskMax int64
+	for _, file := range []string{"/proc/sys/kernel/pid_max", "/proc/sys/kernel/threads-max"} {
+		if content, err := ioutil.ReadFile(file); err == nil {
+			if limit, err := strconv.ParseInt(string(content[:len(content)-1]), 10, 64); err == nil {
+				if taskMax == 0 || taskMax > limit {
+					taskMax = limit
+				}
+			}
 		}
 	}
+	rlimit.MaxPID = &taskMax
```

Comment:
Maybe it'd be more appropriate to retain the meaning of `rlimit.MaxPID` and have a new field for maximum tasks?

Reply:
Then we need to change other places where rlimit.MaxPID is used to use rlimit.MaxTasks. Not sure if it's worthwhile.

I think MaxPID is actually  MaxTasks just as PID in PIDCGroup, which kinda means for the limit of tasks instead of the number of process ID  description.

Reply:
How about commenting the definition of `MaxPID` to mention what it's counting / that the thing it counts might be even more platform-specific than you think.

Reply:
Comments added